lib/chef/property.rb

```diff
@@ -448,18 +452,22 @@ def emit_dsl
       # stack trace if you use `define_method`.
       declared_in.class_eval <<-EOM, __FILE__, __LINE__+1
         def #{name}(value=NOT_PASSED)
+          raise "Property #{name} of \#{self} cannot be passed a block!" if block_given?
```

Comment:
For now, I think I'll explain and let them know they need to rename the property. I'll work on a separate PR letting them use the recipe DSL unhooked from everything else. i.e. @coderanger's point of mixing DSLs; even if we do mix, we need to provide the DSLs separately in at least one form so users can disambiguate if they want.